src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -286,25 +289,56 @@ public void startDetector(
         IndexAnomalyDetectorJobActionHandler handler,
         User user,
         TransportService transportService,
+        ThreadContext.StoredContext context,
         ActionListener<AnomalyDetectorJobResponse> listener
     ) {
+        // upgrade index mapping of AD default indices
+        detectionIndices.update();
+
         getDetector(detectorId, (detector) -> {
             if (!detector.isPresent()) {
                 listener.onFailure(new OpenSearchStatusException(FAIL_TO_FIND_DETECTOR_MSG + detectorId, RestStatus.NOT_FOUND));
                 return;
             }
             if (validateDetector(detector.get(), listener)) { // validate if detector is ready to start
```

Comment:
If validateDetector returns false, would listener return?

Reply:
This is some old code, will return in `validateDetector` , will add more comment.